drivers: sensor: hts221 compilation issue linked to DT property drdy_gpios

**Describe the bug**
While pushing PR #27013, Buildkite highlight a compilation issue in hts221 linked to DT drdy_gpios

**To Reproduce**
Steps to reproduce the behavior:
1. west build -p auto -b disco_l475_iot1 tests/drivers/build_all/ -- -DCONF_FILE=sensors_trigger_a_h.conf
2. See error

**Expected behavior**
Compilation successful

**Logs and console output**
```
FAILED: zephyr/drivers/sensor/hts221/CMakeFiles/drivers__sensor__hts221.dir/hts221.c.obj
--
  | ccache /opt/sdk/zephyr-sdk-0.11.3/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBUILD_VERSION=v2.4.0-rc1-122-gb3ae394ac314 -DCORE_CM4 -DKERNEL -DSTM32L475xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../include -Izephyr/include/generated -I../../../../../../soc/arm/st_stm32/stm32l4 -I../../../../../../drivers -I../../../../../../soc/arm/st_stm32/common -I../../../../../../subsys/testsuite/include -I/var/lib/buildkite-agent/zephyr-module-cache/modules/hal/cmsis/CMSIS/Core/Include -I/var/lib/buildkite-agent/zephyr-module-cache/modules/hal/stm32/stm32cube/stm32l4xx/soc -I/var/lib/buildkite-agent/zephyr-module-cache/modules/hal/stm32/stm32cube/stm32l4xx/drivers/include -I/var/lib/buildkite-agent/zephyr-module-cache/modules/hal/stm32/stm32cube/stm32l4xx/drivers/include/Legacy -isystem ../../../../../../lib/libc/minimal/include -isystem /opt/sdk/zephyr-sdk-0.11.3/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/include -isystem /opt/sdk/zephyr-sdk-0.11.3/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/include-fixed -Os -imacros /workdir/zephyr/sanity-out/disco_l475_iot1/tests/drivers/build_all/sensors.build.trigger_a_h/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -imacros /workdir/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wno-address-of-packed-member -Wno-unused-but-set-variable -Werror=implicit-int -Werror -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/workdir/zephyr/tests/drivers/build_all=CMAKE_SOURCE_DIR -fmacro-prefix-map=/workdir/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/workdir=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT zephyr/drivers/sensor/hts221/CMakeFiles/drivers__sensor__hts221.dir/hts221.c.obj -MF zephyr/drivers/sensor/hts221/CMakeFiles/drivers__sensor__hts221.dir/hts221.c.obj.d -o zephyr/drivers/sensor/hts221/CMakeFiles/drivers__sensor__hts221.dir/hts221.c.obj   -c /workdir/zephyr/drivers/sensor/hts221/hts221.c
  | In file included from ../../../../../../soc/arm/st_stm32/stm32l4/soc.h:30,
  | from ../../../../../../include/arch/arm/aarch32/cortex_m/cmsis.h:17,
  | from ../../../../../../include/arch/arm/aarch32/syscall.h:29,
  | from ../../../../../../include/arch/syscall.h:19,
  | from ../../../../../../include/syscall.h:12,
  | from ../../../../../../include/kernel_includes.h:36,
  | from ../../../../../../include/kernel.h:17,
  | from ../../../../../../include/init.h:11,
  | from ../../../../../../include/device.h:22,
  | from ../../../../../../include/drivers/i2c.h:23,
  | from /workdir/zephyr/drivers/sensor/hts221/hts221.c:9:
  | zephyr/include/generated/devicetree_unfixed.h:3657:31: error: 'DT_N_S_soc_S_i2c_40005800_S_hts221_5f_P_drdy_gpios_IDX_0_VAL_pin' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_i2c_40005800_S_lsm6dsl_6a_P_irq_gpios_IDX_0_VAL_pin'?
  | 3657 \| #define DT_N_INST_0_st_hts221 DT_N_S_soc_S_i2c_40005800_S_hts221_5f
  | \|                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | ../../../../../../include/devicetree.h:1911:38: note: in definition of macro 'DT_CAT'
  | 1911 \| #define DT_CAT(node_id, prop_suffix) node_id##prop_suffix
  | \|                                      ^~~~~~~
  | ../../../../../../include/devicetree.h:647:2: note: in expansion of macro 'DT_PROP'
  | 647 \|  DT_PROP(node_id, pha##_IDX_##idx##_VAL_##cell)
  | \|  ^~~~~~~
  | ../../../../../../include/devicetree/gpio.h:116:2: note: in expansion of macro 'DT_PHA_BY_IDX'
  | 116 \|  DT_PHA_BY_IDX(node_id, gpio_pha, idx, pin)
  | \|  ^~~~~~~~~~~~~
  | ../../../../../../include/devicetree/gpio.h:218:2: note: in expansion of macro 'DT_GPIO_PIN_BY_IDX'
  | 218 \|  DT_GPIO_PIN_BY_IDX(DT_DRV_INST(inst), gpio_pha, idx)
  | \|  ^~~~~~~~~~~~~~~~~~
  | ../../../../../../include/sys/util_internal.h:106:36: note: in expansion of macro 'DT_N_INST_0_st_hts221'
  | 106 \| #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
  | \|                                    ^
  | ../../../../../../include/sys/util_internal.h:105:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  | 105 \| #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
  | \|                          ^~~~~~~~~~~~~~~~~~
  | ../../../../../../include/devicetree.h:297:31: note: in expansion of macro 'UTIL_CAT'
  | 297 \| #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
  | \|                               ^~~~~~~~
  | ../../../../../../include/devicetree.h:1426:27: note: in expansion of macro 'DT_INST'
  | 1426 \| #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
  | \|                           ^~~~~~~
  | ../../../../../../include/devicetree/gpio.h:218:21: note: in expansion of macro 'DT_DRV_INST'
  | 218 \|  DT_GPIO_PIN_BY_IDX(DT_DRV_INST(inst), gpio_pha, idx)
  | \|                     ^~~~~~~~~~~
  | ../../../../../../include/devicetree/gpio.h:229:2: note: in expansion of macro 'DT_INST_GPIO_PIN_BY_IDX'
  | 229 \|  DT_INST_GPIO_PIN_BY_IDX(inst, gpio_pha, 0)
  | \|  ^~~~~~~~~~~~~~~~~~~~~~~
  | /workdir/zephyr/drivers/sensor/hts221/hts221.c:186:14: note: in expansion of macro 'DT_INST_GPIO_PIN'
  | 186 \|  .drdy_pin = DT_INST_GPIO_PIN(0, drdy_gpios),
  | \|              ^~~~~~~~~~~~~~~~
  | zephyr/include/generated/devicetree_unfixed.h:3657:31: error: 'DT_N_S_soc_S_i2c_40005800_S_hts221_5f_P_drdy_gpios_IDX_0_PH_P_label' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_i2c_40005800_S_hts221_5f_P_reg_IDX_0'?
  | 3657 \| #define DT_N_INST_0_st_hts221 DT_N_S_soc_S_i2c_40005800_S_hts221_5f
  | \|                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | ../../../../../../include/devicetree.h:1911:38: note: in definition of macro 'DT_CAT'
  | 1911 \| #define DT_CAT(node_id, prop_suffix) node_id##prop_suffix
  | \|                                      ^~~~~~~
  | ../../../../../../include/devicetree.h:582:2: note: in expansion of macro 'DT_PROP'
  | 582 \|  DT_PROP(DT_PHANDLE_BY_IDX(node_id, phs, idx), prop)
  | \|  ^~~~~~~
  | ../../../../../../include/devicetree.h:393:32: note: in expansion of macro 'DT_CAT'
  | 393 \| #define DT_PROP(node_id, prop) DT_CAT(node_id, _P_##prop)
  | \|                                ^~~~~~
  | ../../../../../../include/devicetree.h:844:2: note: in expansion of macro 'DT_PROP'
  | 844 \|  DT_PROP(node_id, prop##_IDX_##idx##_PH)
  | \|  ^~~~~~~
  | ../../../../../../include/devicetree.h:582:10: note: in expansion of macro 'DT_PHANDLE_BY_IDX'
  | 582 \|  DT_PROP(DT_PHANDLE_BY_IDX(node_id, phs, idx), prop)
  | \|          ^~~~~~~~~~~~~~~~~
  | ../../../../../../include/devicetree/gpio.h:61:2: note: in expansion of macro 'DT_PROP_BY_PHANDLE_IDX'
  | 61 \|  DT_PROP_BY_PHANDLE_IDX(node_id, gpio_pha, idx, label)
  | \|  ^~~~~~~~~~~~~~~~~~~~~~
  | ../../../../../../include/devicetree/gpio.h:195:2: note: in expansion of macro 'DT_GPIO_LABEL_BY_IDX'
  | 195 \|  DT_GPIO_LABEL_BY_IDX(DT_DRV_INST(inst), gpio_pha, idx)
  | \|  ^~~~~~~~~~~~~~~~~~~~
  | ../../../../../../include/sys/util_internal.h:106:36: note: in expansion of macro 'DT_N_INST_0_st_hts221'
  | 106 \| #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
  | \|                                    ^
  | ../../../../../../include/sys/util_internal.h:105:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  | 105 \| #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
  | \|                          ^~~~~~~~~~~~~~~~~~
  | ../../../../../../include/devicetree.h:297:31: note: in expansion of macro 'UTIL_CAT'
  | 297 \| #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
  | \|                               ^~~~~~~~
  | ../../../../../../include/devicetree.h:1426:27: note: in expansion of macro 'DT_INST'
  | 1426 \| #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
  | \|                           ^~~~~~~
  | ../../../../../../include/devicetree/gpio.h:195:23: note: in expansion of macro 'DT_DRV_INST'
  | 195 \|  DT_GPIO_LABEL_BY_IDX(DT_DRV_INST(inst), gpio_pha, idx)
  | \|                       ^~~~~~~~~~~
  | ../../../../../../include/devicetree/gpio.h:205:2: note: in expansion of macro 'DT_INST_GPIO_LABEL_BY_IDX'
  | 205 \|  DT_INST_GPIO_LABEL_BY_IDX(inst, gpio_pha, 0)
  | \|  ^~~~~~~~~~~~~~~~~~~~~~~~~
  | /workdir/zephyr/drivers/sensor/hts221/hts221.c:188:21: note: in expansion of macro 'DT_INST_GPIO_LABEL'
  | 188 \|  .drdy_controller = DT_INST_GPIO_LABEL(0, drdy_gpios),
  | \|                     ^~~~~~~~~~~~~~~~~~
  | [145/208] Building C object zephyr/drivers/sensor/hts221/CMakeFiles/drivers__sensor__hts221.dir/hts221_trigger.c.obj
  | [146/208] Building C object zephyr/drivers/spi/CMakeFiles/drivers__spi.dir/spi_ll_stm32.c.obj
  | ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS:  Linux and Windows
 - Toolchain Zephyr SDK
 - Commit SHA: 6c53e9331854a3485112685c47b77852f3576224

--- a/drivers/sensor/hts221/hts221.c
+++ b/drivers/sensor/hts221/hts221.c
@@ -106,7 +106,7 @@ static int hts221_read_conversion_data(const struct device *dev)
 }
 
 static const struct sensor_driver_api hts221_driver_api = {
-#if CONFIG_HTS221_TRIGGER
+#if HTS221_TRIGGER_ENABLED
 	.trigger_set = hts221_trigger_set,
 #endif
 	.sample_fetch = hts221_sample_fetch,
@@ -168,11 +168,13 @@ int hts221_init(const struct device *dev)
 		return -EINVAL;
 	}
 
-#ifdef CONFIG_HTS221_TRIGGER
+#if HTS221_TRIGGER_ENABLED
 	if (hts221_init_interrupt(dev) < 0) {
 		LOG_ERR("Failed to initialize interrupt.");
 		return -EIO;
 	}
+#else
+	LOG_INF("Cannot enable trigger without drdy-gpios");
 #endif
 
 	return 0;
@@ -182,11 +184,11 @@ static struct hts221_data hts221_driver;
 static const struct hts221_config hts221_cfg = {
 	.i2c_bus = DT_INST_BUS_LABEL(0),
 	.i2c_addr = DT_INST_REG_ADDR(0),
-#ifdef CONFIG_HTS221_TRIGGER
+#if HTS221_TRIGGER_ENABLED
 	.drdy_pin = DT_INST_GPIO_PIN(0, drdy_gpios),
 	.drdy_flags = DT_INST_GPIO_FLAGS(0, drdy_gpios),
 	.drdy_controller = DT_INST_GPIO_LABEL(0, drdy_gpios),
-#endif /* CONFIG_HTS221_TRIGGER */
+#endif /* HTS221_TRIGGER_ENABLED */
 };
 
 DEVICE_AND_API_INIT(hts221, DT_INST_LABEL(0), hts221_init,

--- a/drivers/sensor/hts221/hts221.h
+++ b/drivers/sensor/hts221/hts221.h
@@ -12,6 +12,9 @@
 #include <zephyr/types.h>
 #include <drivers/gpio.h>
 
+#define HTS221_TRIGGER_ENABLED (DT_INST_NODE_HAS_PROP(0, drdy_gpios) && \
+				IS_ENABLED(CONFIG_HTS221_TRIGGER))
+
 #define HTS221_AUTOINCREMENT_ADDR	BIT(7)
 
 #define HTS221_REG_WHO_AM_I		0x0F
@@ -42,7 +45,7 @@ struct hts221_data {
 	int16_t t0_out;
 	int16_t t1_out;
 
-#ifdef CONFIG_HTS221_TRIGGER
+#if HTS221_TRIGGER_ENABLED
 	const struct device *dev;
 	const struct device *drdy_dev;
 	struct gpio_callback drdy_cb;
@@ -58,20 +61,20 @@ struct hts221_data {
 	struct k_work work;
 #endif
 
-#endif /* CONFIG_HTS221_TRIGGER */
+#endif /* HTS221_TRIGGER_ENABLED */
 };
 
 struct hts221_config {
 	const char *i2c_bus;
 	uint16_t i2c_addr;
-#ifdef CONFIG_HTS221_TRIGGER
+#if HTS221_TRIGGER_ENABLED
 	gpio_pin_t drdy_pin;
 	gpio_flags_t drdy_flags;
 	const char *drdy_controller;
-#endif /* CONFIG_HTS221_TRIGGER */
+#endif /* HTS221_TRIGGER_ENABLED */
 };
 
-#ifdef CONFIG_HTS221_TRIGGER
+#if HTS221_TRIGGER_ENABLED
 int hts221_trigger_set(const struct device *dev,
 			const struct sensor_trigger *trig,
 			sensor_trigger_handler_t handler);

--- a/drivers/sensor/hts221/hts221_trigger.c
+++ b/drivers/sensor/hts221/hts221_trigger.c
@@ -4,6 +4,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define DT_DRV_COMPAT st_hts221
+
 #include <device.h>
 #include <drivers/i2c.h>
 #include <sys/__assert.h>
@@ -13,6 +15,7 @@
 #include <logging/log.h>
 #include "hts221.h"
 
+#if HTS221_TRIGGER_ENABLED
 LOG_MODULE_DECLARE(HTS221, CONFIG_SENSOR_LOG_LEVEL);
 
 static inline void setup_drdy(const struct device *dev,
@@ -163,3 +166,4 @@ int hts221_init_interrupt(const struct device *dev)
 
 	return 0;
 }
+#endif /* HTS221_TRIGGER_ENABLED */

